src/app/clusters/window-covering-server/window-covering-server.cpp

```diff
@@ -0,0 +1,685 @@
+    uint8_t lift   = OperationalStateToValue(status.lift);
+    uint8_t tilt   = OperationalStateToValue(status.tilt);
+    uint8_t value  = (global & 0x03) | ((lift & 0x03) >> 2) | ((tilt & 0x03) >> 4);
+    writeAttribute(mEndPoint, ZCL_WC_OPERATIONAL_STATUS_ATTRIBUTE_ID, &value, ZCL_BITMAP8_ATTRIBUTE_TYPE);
+}
+
+const WindowCover::OperationalStatus WindowCover::OperationalStatusGet(void)
+{
+    uint8_t value = 0;
+    OperationalStatus status;
+
+    readAttribute(mEndPoint, ZCL_WC_OPERATIONAL_STATUS_ATTRIBUTE_ID, &value, sizeof(uint8_t));
+    status.global = ValueToOperationalState(value & 0x03);
+    status.lift   = ValueToOperationalState((value << 2) & 0x03);
+    status.tilt   = ValueToOperationalState((value << 4) & 0x03);
+    return status;
+}
+
+void WindowCover::EndProductTypeSet(EmberAfWcEndProductType type)
+{
+    writeAttribute(mEndPoint, ZCL_WC_END_PRODUCT_TYPE_ATTRIBUTE_ID, static_cast<uint8_t *> &type, ZCL_ENUM8_ATTRIBUTE_TYPE);
+}
+
+EmberAfWcEndProductType WindowCover::EndProductTypeGet(void)
+{
+    EmberAfWcEndProductType type;
+    readAttribute(mEndPoint, ZCL_WC_END_PRODUCT_TYPE_ATTRIBUTE_ID, static_cast<uint8_t *> &type, sizeof(uint8_t));
+    return type;
+}
+
+void WindowCover::ModeSet(WindowCover::Mode mode)
+{
+    uint8_t value = (mode.motorDirReversed ? 0x01 : 0) | (mode.calibrationMode ? 0x02 : 0) | (mode.maintenanceMode ? 0x04 : 0) |
+        (mode.ledDisplay ? 0x08 : 0);
+    writeAttribute(mEndPoint, ZCL_WC_MODE_ATTRIBUTE_ID, &value, ZCL_BITMAP8_ATTRIBUTE_TYPE);
+}
+
+const WindowCover::Mode WindowCover::ModeGet(void)
+{
+    uint8_t value = 0;
+    Mode mode;
+    writeAttribute(mEndPoint, ZCL_WC_MODE_ATTRIBUTE_ID, &value, sizeof(uint8_t));
+    mode.motorDirReversed = (value & 0x01) ? 1 : 0;
+    mode.calibrationMode  = (value & 0x02) ? 1 : 0;
+    mode.maintenanceMode  = (value & 0x04) ? 1 : 0;
+    mode.ledDisplay       = (value & 0x08) ? 1 : 0;
+    return mode;
+}
+
+void WindowCover::SafetyStatusSet(WindowCover::SafetyStatus status)
+{
+    uint16_t value = (status.remoteLockout ? 0x0001 : 0) | (status.tamperDetection ? 0x0002 : 0) |
+        (status.failedCommunication ? 0x0004 : 0) | (status.positionFailure ? 0x0008 : 0) |
+        (status.thermalProtection ? 0x0010 : 0) | (status.obstacleDetected ? 0x0020 : 0) | (status.powerIssue ? 0x0040 : 0) |
+        (status.stopInput ? 0x0080 : 0);
+    value |= (uint16_t)(status.motorJammed ? 0x0100 : 0) | (uint16_t)(status.hardwareFailure ? 0x0200 : 0) |
+        (uint16_t)(status.manualOperation ? 0x0400 : 0);
+    writeAttribute(mEndPoint, ZCL_WC_OPERATIONAL_STATUS_ATTRIBUTE_ID, static_cast<uint8_t *> &value, ZCL_BITMAP16_ATTRIBUTE_TYPE);
+}
+
+const WindowCover::SafetyStatus WindowCover::SafetyStatusGet(void)
+{
+    uint16_t value = 0;
+    WindowCover::SafetyStatus status;
+
+    readAttribute(mEndPoint, ZCL_WC_OPERATIONAL_STATUS_ATTRIBUTE_ID, static_cast<uint8_t *> &value, sizeof(uint16_t));
```

Comment:
```suggestion
    readAttribute(mEndPoint, ZCL_WC_OPERATIONAL_STATUS_ATTRIBUTE_ID, static_cast<uint8_t *>(&value), sizeof(uint16_t));
```

Reply:
Replaced with the corresponding Accessors.h function.